### PR TITLE
lib/posix-process: Solve redefining conflicts

### PR DESCRIPTION
**Pull request by dragosargint**
There are many functions that are defined in musl as well as in posix-process.
When using musl we want to use the musl functions. When we use musl the
`LIBSYSCALL_SHIM_NOWRAPPER` macro is set, which leads to setting of
`UK_LIBC_SYSCALLS` to 0. We can use this to exclude the redefined functions
from posix-process.

This PR is part of a series:
https://github.com/unikraft/unikraft/pull/454
https://github.com/unikraft/unikraft/pull/453
https://github.com/unikraft/unikraft/pull/442
https://github.com/unikraft/unikraft/pull/441

Signed-off-by: Dragos Iulian Argint <dragosargint21@gmail.com>

### CODE REVIEW

**Review by mogasergiu — APPROVED** [review]
Looks good to me.

Reviewed-by: Sergiu Moga <sergiu.moga@protonmail.com>

File: lib/posix-process/process.c [hunk 1]
```diff
@@ -281,19 +286,30 @@ int tcsetpgrp(int fd __unused, pid_t pgrp)
...
+	/* We don't support priority updates for unikernels */
+	return -EPERM;
+}
 
+#if UK_LIBC_SYSCALLS
 int nice(int inc __unused)
 {
 	/* We don't support priority updates for unikernels */
-	errno = EPERM;
+	errno = uk_syscall_r_nice(inc);
```
Comment (skuenzer) [inline lib/posix-process/process.c]: Hum, this call is making the errno value negative. Secondly, `UK_SYSCALL_R_DEFINE(int, nice, int, inc)` is already provide `nice` so this wrapper should be removed completely.

**Review by skuenzer — CHANGES_REQUESTED** [review]
Hey Dragos,

can you try to declare all calls with `UK_SYSCALL_R_DEFINE` instead? Then we register them properly to the syscall shim and don't just remove the C symbol if a richt libc is is part of the build.

Thanks,

Simon

File: lib/posix-process/process.c (hunk 1 above)
Comment (dragosargint) [reply lib/posix-process/process.c]: I agree on the `nice` syscall, but the other functions are not syscalls. For example `wait3` is defined in musl as follows:
```
pid_t wait3(int *status, int options, struct rusage *usage)
{
        return wait4(-1, status, options, usage);
}
```

Comment (skuenzer) [reply lib/posix-process/process.c]: Fair enough, this makes sense. In fact `wait`, `waitpid` and `wait3` are all mapped to `wait4`.

**Review by skuenzer — APPROVED** [review]
Approved-by: Simon Kuenzer <simon.kuenzer@neclab.eu>

**Pull request closed without merging**